### PR TITLE
Fix bad assertion on malformed QUIC trailer

### PR DESCRIPTION
**Pull request by pksohn**
Commit Message: Fix bad assertion on malformed QUIC trailer
Additional Description: This change allows for a graceful stream reset instead of failing an assertion on a malformed QUIC trailer (e.g. uppercase or empty name). Found via fuzz testing by robert.femmer@x41-dsec.de.
Risk Level: Low
Testing: Includes unit test

### CODE REVIEW

**Review by alyssawilk — COMMENTED** [review]
Sorry for the delay. LGTM but is it possible to add an e2e test or will client side validation result in us not being able to send the invali headers?

File: test/integration/quic_http_integration_test.cc [hunk 1]
```diff
@@ -1390,6 +1390,28 @@ TEST_P(QuicHttpIntegrationTest, DeferredLoggingWithInternalRedirect) {
   EXPECT_EQ(/* RESP(test-header) */ metrics.at(21), "-");
 }
 
+TEST_P(QuicHttpIntegrationTest, InvalidTrailer) {
+  initialize();
+  // Empty string in trailer key is invalid.
+  Http::TestRequestTrailerMapImpl request_trailers{{"", "foo"}};
+  Http::TestResponseTrailerMapImpl response_trailers{{"a", "b"}, {"c", "d"}};
```
Comment (alyssawilk) [inline test/integration/quic_http_integration_test.cc:1397]: Awesome, thanks for tracking down a testable case!
Do we need response trailers or can we nix them?

Comment (pksohn) [reply test/integration/quic_http_integration_test.cc]: Meant to remove those, thanks for the catch!

**Review by alyssawilk — APPROVED** [review]

**Comment by alyssawilk** [comment]
it's in the submit queue so if CI flakes just retest until it lands :-)

**Pull request merged**